### DISCUSSION TITLE
How to use <keep-alive> ?

### DISCUSSION THREAD

**Discussion by webfansplz** [nuxt/nuxt#17327] [Questions] [answered]
https://github.com/nuxt/framework/blob/23397e603c97b3a5b75b035ce72dbc633bbb13a5/packages/nuxt3/src/pages/runtime/page.vue#L5

Does Nuxt3 support `<keep-alive>` now ? 

It doesn't seem to be supported yet .

**Comment by xiaojieajie** [comment, marked as answer]
Already supported

Add to pages that need to be cached

<script lang="ts" setup >
  definePageMeta({
    keepalive: {},
  });
</script>

**Reply by baymax-arch** [reply]
这样从详情页返回，为啥会执行onmouted跟setup呢，nuxt3 keepalive的书写方式就加上definePageMeta({ keepalive: {}, })吗，还需要其他操作吗，好家伙看了一下你的个人网站，这个问题好像在掘金里问过你

**Reply by xiaojieajie** [reply]
> 这样从详情页返回，为啥会执行onmouted跟setup呢，nuxt3 keepalive的书写方式就加上definePageMeta({ keepalive: {}, })吗，还需要其他操作吗，好家伙看了一下你的个人网站，这个问题好像在掘金里问过你

那你的文章列表页需要exclude下 详情页

**Reply by baymax-arch** [reply]
刚才找到一个答案，只有详情页同时开启keepalive的情况下再返回列表页才会保持原来的状态，试了一下，果然是这样，就是不知道咋解决

**Reply by taoxhsmile** [reply]
有nuxt3交流群吗，搞个群方便聊一些问题，我 微 信  q739731324

**Reply by xiaojieajie** [reply]
> 有nuxt3交流群吗，搞个群方便聊一些问题，我 微 信 q739731324

那我创建一个群吧 哈哈哈

**Reply by 11003** [reply]
> q739731324

能说一下QQ群吗？我想问一下关于 keepalive 的问题。我也遇到了坑；去文章详情，然后出来。没有定位到之前滑到的位置那儿

**Reply by baymax-arch** [reply]
不知道有没有群，你试着在详情页和列表页同时开启keepalive> > q739731324
> 
> 
> 
> 能说一下QQ群吗？我想问一下关于 keepalive 的问题。我也遇到了坑；去文章详情，然后出来。没有定位到之前滑到的位置那儿
